trio/_file_io/_path.py

```diff
@@ -0,0 +1,138 @@
+        type(cls).generate_wraps(cls, attrs)
+        type(cls).generate_magic(cls, attrs)
+
+    def generate_forwards(cls, attrs):
+        # forward functions of _forwards
+        for attr_name, attr in cls._forwards.__dict__.items():
+            if attr_name.startswith('_') or attr_name in attrs:
+                continue
+
+            if isinstance(attr, property):
+                cls._forward.append(attr_name)
+            elif isinstance(attr, types.FunctionType):
+                wrapper = _forward_factory(cls, attr_name, attr)
+                setattr(cls, attr_name, wrapper)
+            else:
+                raise TypeError(attr_name, type(attr))
+
+    def generate_wraps(cls, attrs):
+        # generate wrappers for functions of _wraps
+        for attr_name, attr in cls._wraps.__dict__.items():
+            if attr_name.startswith('_') or attr_name in attrs:
+                continue
+
+            if isinstance(attr, classmethod):
+                setattr(cls, attr_name, attr)
+            elif isinstance(attr, types.FunctionType):
+                wrapper = thread_wrapper_factory(cls, attr_name)
+                setattr(cls, attr_name, wrapper)
+            else:
+                raise TypeError(attr_name, type(attr))
+
+    def generate_magic(cls, attrs):
+        # generate wrappers for magic
+        for attr_name in cls._forward_magic:
+            attr = getattr(cls._forwards, attr_name)
+            wrapper = _forward_factory(cls, attr_name, attr)
+            setattr(cls, attr_name, wrapper)
+
+
+class AsyncPath(metaclass=AsyncAutoWrapperType):
+    """:class:`trio.AsyncPath` is a :class:`~pathlib.Path` wrapper executes concrete
+    non-computational Path methods in :meth:`trio.run_in_worker_thread`.
+
+    """
+
+    _wraps = Path
+    _forwards = PurePath
+    _forward_magic = [
+        '__str__', '__bytes__',
+        '__eq__', '__lt__', '__le__', '__gt__', '__ge__'
+    ]
+
+    def __new__(cls, *args, **kwargs):
+        path = Path(*args, **kwargs)
+
+        self = cls._from_wrapped(path)
+        return self
+
+    def __getattr__(self, name):
+        if name in self._forward:
+            return getattr(self._wrapped, name)
+        raise AttributeError(name)
+
+    def __dir__(self):
+        return super().__dir__() + self._forward
+
+    def __repr__(self):
+        return 'AsyncPath({})'.format(str(self))
+
+    def __fspath__(self):
+        try:
+            return self._wrapped.__fspath__()
+        except AttributeError:
+            return str(self)
+
+    @classmethod
+    def _from_wrapped(cls, wrapped):
+        self = object.__new__(cls)
+        self._wrapped = wrapped
+        return self
+
+    async def open(self, *args, **kwargs):
+        func = partial(self._wrapped.open, *args, **kwargs)
+        value = await trio.run_in_worker_thread(func)
+        return trio.wrap_file(value)
+
+
+# not documented upstream
+delattr(AsyncPath.absolute, '__doc__')
```

Comment:
The value of `trio.AsyncPath.absolute.__doc__` makes a reference to `:meth:~pathlib.Path.absolute`, which does not exist. If this line were removed, building docs would fail.